src/js/player.js

```diff
@@ -1643,6 +1645,34 @@ class Player extends Component {
     }
   }
 
+  /**
+   * Handle a double-click on the media element to enter/exit fullscreen
+   *
+   * @param {EventTarget~Event} event
+   *        the event that caused this function to trigger
+   *
+   * @listens Tech#dblclick
+   * @private
+   */
+  handleTechDoubleClick_(event) {
+    if (!this.controls_) {
+      return;
+    }
+
+    const inAllowedEls = Array.prototype.some.call(
+        this.$$('.vjs-control-bar, .vjs-modal-dialog'),
+        el => el.contains(event.target)
+      );
+
+    if (!inAllowedEls) {
```

Comment:
This seems confusing. If it's not in allowed elements it's prohibited?

Reply:
maybe we just need a comment? Basically, we shouldn't try toggling fullscreen if we double click inside of the control bar or the modal dialog.

Reply:
Added a comment.